packages/math/src/integers.ts

```diff
@@ -227,45 +228,55 @@ export class Uint64 implements Integer, WithByteConverters {
       throw new Error("Input is not an integer");
     }
 
-    let bigint: BN;
-    try {
-      bigint = new BN(input);
-    } catch {
+    if (!Number.isSafeInteger(input)) {
       throw new Error("Input is not a safe integer");
     }
+
+    const bigint = BigInt(input);
     return new Uint64(bigint);
   }
 
-  private readonly data: BN;
+  private readonly data: bigint;
 
-  private constructor(data: BN) {
-    if (data.isNeg()) {
+  private constructor(data: bigint) {
+    if (data < 0n) {
       throw new Error("Input is negative");
     }
-    if (data.gt(uint64MaxValue)) {
+    if (data > uint64MaxValue) {
       throw new Error("Input exceeds uint64 range");
     }
     this.data = data;
   }
 
   public toBytesBigEndian(): Uint8Array {
-    return Uint8Array.from(this.data.toArray("be", 8));
+    return this.toBytesLittleEndian().reverse();
   }
 
   public toBytesLittleEndian(): Uint8Array {
-    return Uint8Array.from(this.data.toArray("le", 8));
+    const bytes = new Uint8Array(8);
+    let value = this.data;
+    for (let i = 0; i < bytes.length; i++) {
+      bytes[i] = Number(value % 256n);
+      value /= 256n;
+    }
+
+    return bytes;
   }
 
   public toString(): string {
     return this.data.toString(10);
   }
 
   public toBigInt(): bigint {
-    return BigInt(this.toString());
+    return this.data;
   }
 
   public toNumber(): number {
-    return this.data.toNumber();
+    const num = Number(this.data);
+    if (!Number.isSafeInteger(num)) {
```

Comment:
Let's do this check before the conversion to be on the safe side of JS magic. Something along the lines of

```ts
if (this.data > BigInt(Number.MAX_SAFE_INTEGER)) {
  // ...
}
```